Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5165,7 +5165,30 @@ void WebPageProxy::findStringMatches(const String& string, OptionSet<FindOptions
 
 void WebPageProxy::findString(const String& string, OptionSet<FindOptions> options, unsigned maxMatchCount, CompletionHandler<void(bool)>&& callbackFunction)
 {
-    sendWithAsyncReply(Messages::WebPage::FindString(string, options, maxMatchCount), WTFMove(callbackFunction));
+    class FindStringCallbackAggregator : public RefCounted<FindStringCallbackAggregator> {
+    public:
+        static Ref<FindStringCallbackAggregator> create(CompletionHandler<void(bool)>&& completionHandler) { return adoptRef(*new FindStringCallbackAggregator(WTFMove(completionHandler))); }
+        void wasFound(bool result) { m_found |= result; }
+        ~FindStringCallbackAggregator()
+        {
+            m_completionHandler(m_found);
+        }
+    private:
+        explicit FindStringCallbackAggregator(CompletionHandler<void(bool)>&& completionHandler)
+            : m_completionHandler(WTFMove(completionHandler))
+            , m_found(false)
+        {
+        }
+        CompletionHandler<void(bool)> m_completionHandler;
+        bool m_found;
```

Comment:
This needs to be initialized to false.

Reply:
It's initialized by the constructor, but I'll move it here.